dspace-uclouvain/src/main/java/org/dspace/uclouvain/consumer/DegreeMetadataConsumer.java

```diff
@@ -0,0 +1,215 @@
+package org.dspace.uclouvain.consumer;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.event.Consumer;
+import org.dspace.event.Event;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.core.model.MetadataField;
+import org.dspace.uclouvain.degreeMappers.DegreeMappersService;
+import org.dspace.uclouvain.degreeMappers.exceptions.GeneratorProcessException;
+import org.dspace.uclouvain.degreeMappers.factory.DegreeMappersServiceFactory;
+import org.dspace.uclouvain.degreeMappers.model.DegreeMapper;
+
+/**
+ * Consumer to generate additional metadata from the degree code metadata field.
+ * This is using the DegreeMappersService to get the additional metadata from a configuration file.
+ *
+ * Authored-By: Michaël Pourbaix <michael.pourbaix@uclouvain.be>
+ */
+public class DegreeMetadataConsumer implements Consumer {
+
+    private MetadataField degreeCodeField;
+    private MetadataField rootDegreeCodeField;
+    private MetadataField rootDegreeLabelField;
+    private MetadataField facultyCodeField;
+    private MetadataField facultyNameField;
+    
+    private ItemService itemService;
+    private DegreeMappersService degreeMappersService;
+    private ConfigurationService configurationService;
+
+    @Override
+    public void initialize() throws Exception {
+        this.configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        this.itemService = ContentServiceFactory.getInstance().getItemService();
+        this.degreeMappersService = DegreeMappersServiceFactory.getInstance().getDegreeMappersService();
+
+        this.degreeCodeField = new MetadataField(this.configurationService.getProperty("uclouvain.global.metadata.degreecode.field"));
+        this.rootDegreeCodeField = new MetadataField(this.configurationService.getProperty("uclouvain.global.metadata.rootdegreecode.field"));
+        this.rootDegreeLabelField = new MetadataField(this.configurationService.getProperty("uclouvain.global.metadata.rootdegreelabel.field"));
+        this.facultyCodeField = new MetadataField(this.configurationService.getProperty("uclouvain.global.metadata.facultycode.field"));
+        this.facultyNameField = new MetadataField(this.configurationService.getProperty("uclouvain.global.metadata.facultyname.field"));
```

Comment:
If some properties doesn't exist into configuration, it should raise an exception ; otherwise the `consume` method will failed.